### PR TITLE
Default channel for CRDs documentation

### LINKED ISSUE #1618: Document new Channel CRDs Default Mechanism
## What is changing? (Please include as many details as possible.)
Introduced a new ConfigMap for setting up default Channel CRDs. We need to properly document it. 

## How will this impact our users?
v0.9 will remove all Channel provisioner-based stuff, so they will have to switch to Channel CRDs. Once they do that, they will need a way to configure defaults. When doing so, they will need to understand how to properly do that configuration, i.e., they need documentation.

As we are not removing anything in v0.8, they will not be immediately impacted.

## In what release will this take happen (to the best of your knowledge)?
v0.8

## Context
Link to associated PRs or issues from other repos here.

1. https://github.com/knative/eventing/pull/1560
1. https://github.com/knative/eventing/issues/1343

## Additional info
In 0.7 we've introduced the new Channel CRDs model. In 0.8 we have a new defaulting mechanism for these Channel CRDs. We should properly document it, with few examples.

### PR DESCRIPTION
**Pull request by nachocano**
Fixes #1618

## Proposed Changes

- Added a new default-channels-crds.md to explain how the new default mechanism for channels work.

### CODE REVIEW

File: docs/eventing/channels/default-channels-crds.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
+title: "Default channels"
```
Comment (nachocano) [inline docs/eventing/channels/default-channels-crds.md]: There is a defaults-channels.md which should be marked as deprecated in 0.8, and maybe point to this one? What is the procedure to properly do that?

Comment (samodell) [reply docs/eventing/channels/default-channels-crds.md]: @nachocano  We don't really have a mechanism for redirects right now, as far as I know, but I'll check with Richie when he gets back from vacation. I think what we'll do is delete the deprecated file from the 0.8 release; it'll live on in the <0.7 versions.

File: docs/eventing/channels/default-channels-crds.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        spec:
+          numPartitions: 2
+          replicationFactor: 1
+```
+
+Namespace-specific defaults take precedence when matched. In the above example, a
+`Channel` created in the `some-namespace` namespace will be backed by an underlying `KafkaChannel`, 
+not an `InMemoryChannel`.
+
+## Defaults only apply on channel creation
```
Comment (vaikas-google) [inline docs/eventing/channels/default-channels-crds.md]: and broker / sequence creation?

File: docs/eventing/channels/default-channels-crds.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
+---
+title: "Default channels"
+#linkTitle: "OPTIONAL_ALTERNATE_NAV_TITLE"
+weight: 2
+type: "docs"
+---
+
+The default channel configuration allows channels to be created without
```
[the remark this answers is not in the document]
Comment (nachocano) [reply docs/eventing/channels/default-channels-crds.md]: Makes sense, I did some changes... Might still need another iteration.
Thanks for the comment

File: docs/eventing/channels/default-channels-crds.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+specifying an underlying implementation. This leaves the selection of the channel implementation 
+and properties up to the operator. The operator controls the default settings via a
+`ConfigMap`.
+
+## Creating a default channel
```
[the remark this answers is not in the document]
Comment (nachocano) [reply docs/eventing/channels/default-channels-crds.md]: done, makes sense... it was a copy/paste thing from the current doc...

File: docs/eventing/channels/default-channels-crds.md (hunk 2 above)
Comment (nachocano) [reply docs/eventing/channels/default-channels-crds.md]: Added it! I guess I wanted to have this as generic as possible. And add that to broker and sequence documentation. If later on we add choice, etc., etc., and they all use this default channel thing, I didn't want to keep on updating this file, but rather refer to it from the other docs. 
But I think is also fine to add it. Thanks for the comment.

**Comment by vaikas-google** [comment]
/approve

Leaving lgtm for @samodell

File: docs/eventing/channels/default-channels-crds.md (hunk 1 above)
Comment (samodell) [reply docs/eventing/channels/default-channels-crds.md]: @RichieEscarez

File: docs/eventing/channels/default-channels-crds.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+type: "docs"
+---
+
+The default channel configuration allows channels to be created without
+specifying an underlying implementation. This leaves the selection of the channel implementation 
+and properties up to the operator. The operator controls the default settings via a
+`ConfigMap`. For example, when `Brokers` or `Sequences` are created, an operator can configure a default channel to use 
+for their underlying channel-based implementations.
+
+Even though this default channel mechanism aims to ease the usability of the system, users can still create their own
```
Comment (samodell) [inline docs/eventing/channels/default-channels-crds.md]: I think we should describe what a channel does here.  Also, say why someone might want to use a default channel. Right now, we say the default channel "allows channels to be created without specifying an underlying implementation." I'd like us to clarify why someone wouldn't want to specify the implementation to begin with. @nachocano

Comment (samodell) [reply docs/eventing/channels/default-channels-crds.md]: @nachocano  let me know if I should throw some time on your calendar to discuss!

Comment (nachocano) [reply docs/eventing/channels/default-channels-crds.md]: Sam, sorry, I've been busy with other stuff. The comment makes sense to me. Let me address it, and I'll get back to you! Thanks!

Comment (nachocano) [reply docs/eventing/channels/default-channels-crds.md]: Done, please take another look!

**Comment by nachocano** [comment]
@samodell BTW, let me know if you prefer to remove the default-channels.md as it is deprecated, and I can rename this file to have that name. Or you prefer to add a Deprecation note in the other one, and point to this one?

**Pull request merged**